skimage/registration/_lddmm.py

```diff
@@ -0,0 +1,1247 @@
+    # Validate velocity_field_resolution.
+    velocity_field_resolution = _validate_resolution(velocity_field_resolution, velocity_fields.ndim - 2)
+    # Validate affine.
+    affine = _validate_ndarray(affine, required_ndim=2, reshape_to_shape=(len(template_shape) + 1, len(template_shape) + 1))
+    # Verify deform_to.
+    if not isinstance(deform_to, str):
+        raise TypeError(f"deform_to must be of type str.\n"
+            f"type(deform_to): {type(deform_to)}.")
+    elif deform_to not in ["template", "target"]:
+        raise ValueError(f"deform_to must be either 'template' or 'target'.")
+
+    # Compute intermediates.
+    num_timesteps = velocity_fields.shape[-2]
+    delta_t = 1 / num_timesteps
+    template_axes = _compute_axes(template_shape, template_resolution)
+    template_coords = _compute_coords(template_shape, template_resolution)
+    target_axes = _compute_axes(target_shape, target_resolution)
+    target_coords = _compute_coords(target_shape, target_resolution)
+
+    # Create position field.
+    if deform_to == "template":
+        phi = np.copy(template_coords)
+    elif deform_to == "target":
+        phi_inv = np.copy(template_coords)
+
+    # Integrate velocity field.
+    for timestep in (reversed(range(num_timesteps)) if deform_to == "template" else range(num_timesteps)):
+        if deform_to == "template":
+            sample_coords = template_coords + velocity_fields[..., timestep, :] * delta_t
+            phi = interpn(
+                points=template_axes,
+                values=phi - template_coords,
+                xi=sample_coords,
+                bounds_error=False,
+                fill_value=None,
+            ) + sample_coords
+        elif deform_to == "target":
+            sample_coords = template_coords - velocity_fields[..., timestep, :] * delta_t
+            phi_inv = interpn(
+                points=template_axes,
+                values=phi_inv - template_coords,
+                xi=sample_coords,
+                bounds_error=False,
+                fill_value=None,
+            ) + sample_coords
+
+    # Apply the affine transform to the position field.
+    if deform_to == "template":
+        # Apply the affine by multiplication.
+        affine_phi = _multiply_coords_by_affine(affine, phi)
+        # affine_phi has the resolution of the template.
+    elif deform_to == "target":
+        # Apply the affine by interpolation.
+        sample_coords = _multiply_coords_by_affine(inv(affine), target_coords)
+        phi_inv_affine_inv = interpn(
+            points=template_axes,
+            values=phi_inv - template_coords,
+            xi=sample_coords,
+            bounds_error=False,
+            fill_value=None,
+        ) + sample_coords
+        # phi_inv_affine_inv has the resolution of the target.
+
+    # return appropriate position field.
+    if deform_to == "template":
+        return affine_phi
+    elif deform_to == "target":
+        return phi_inv_affine_inv
```

Comment:
For binary choices, I prefer to express them as a boolean. Here I would use something like `inverse_transform={False,True}`, maybe? `inverse=True` would correspond to `deform_to='target'`.

Reply:
I like that actually. Although now these have been deprecated here.